naming/src/main/java/com/alibaba/nacos/naming/core/Instances.java

```diff
@@ -84,8 +86,10 @@ private String recalculateChecksum() {
         }
 
         if (MESSAGE_DIGEST != null) {
-            checksum =
-                new BigInteger(1, MESSAGE_DIGEST.digest((sb.toString()).getBytes(Charset.forName("UTF-8")))).toString(16);
+            synchronized (LOCK) {
```

Comment:
Can ThreadLocal be used to avoid lock contention

Reply:
okay

Reply:
> Can ThreadLocal be used to avoid lock contention

done